.github/workflows/main.yml

```diff
@@ -0,0 +1,59 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - dev
+      - fixes/*
+      - features/*
+
+  pull_request:
+#    branches: [ dev ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ 3.6, 3.7, 3.8 ]
```

Comment:
Do we really need all three?

Reply:
I don't think so, but I am not sure what the version specification does. I have added python 3.6 because this is the version I use.

Reply:
It means that github actions will build the environments for these three python versions to run the checks. Many builds mean more time and energy are consumed. I dropped python 3.6 again and reset some settings in 8d1f207 to reduce the number of builds to two.